src/Result.php

```diff
@@ -16,13 +18,17 @@ public function isValid(): bool
         return $this->errors === [];
     }
 
-    public function addError(string $message): void
+    public function addError(string $message, $key = null): void
```

Comment:
Left like that because `$key` can also be `int`. But passed `$key` is always `string` though, you are right. Corrected.

Reply:
:heavy_check_mark:

Reply:
It actually can be `int` with combinations of `Nested` and `Each`, so I am reverting this back. I can't add `mixed` here because it's only possible since PHP 8. Specifying multiple types is also not possible as far as I remember. At least I can add this to PHPDoc docblock.

Reply:
I think it's fine as is.